app/static/js/app.js

```diff
@@ -124,6 +125,9 @@ function sendKeystroke(keystroke) {
 function onSocketConnect() {
   connectedToServer = true;
   document.getElementById("connection-indicator").connected = true;
+  if (document.getElementById("shutdown-wait").show) {
+    location.reload();
+  }
```

Comment:
Oh, this is clever. I held off on detecting when restart would be complete because I figured it would be a hassle to keep pinging the backend. I forgot that the websocket was already attempting reconnects.